timemachine/cpp/src/harmonic_angle.cu

```diff
@@ -84,17 +84,19 @@ void HarmonicAngle<RealType>::execute_device(
 
     auto start = std::chrono::high_resolution_clock::now();
 
-    k_harmonic_angle_inference<RealType, 3><<<blocks, tpb, 0, stream>>>(
-        A_,
-        d_x,
-        d_p,
-        d_angle_idxs_,
-        d_du_dx,
-        d_du_dp,
-        d_u
-    );
-
-    gpuErrchk(cudaPeekAtLastError());
+    if(A_ > 0) {
```

Comment:
To clarify, this is checking that the number of angle terms is > 0 (since num angles = 0 for a HarmonicAngle force became a more common possibility now that we're doing single-topology), correct?

Do we need to add a similar check for all other valence terms? (I noticed one has been added for HarmonicBonds, but I'm not sure if one has been added for PeriodicTorsions...)

Reply:
This check was always there for PeriodicTorsions - its only the Angle/Bond terms that were missing